tool/src/org/antlr/v4/codegen/UnicodeEscapes.java

```diff
@@ -8,31 +8,42 @@
 
 /**
  * Utility class to escape Unicode code points using various
- * languages' syntaxes.
+ * languages' syntax.
  */
-public abstract class UnicodeEscapes {
-	static public void appendJavaStyleEscapedCodePoint(int codePoint, StringBuilder sb) {
-		if (Character.isSupplementaryCodePoint(codePoint)) {
-			// char is not an 'integral' type, so we have to explicitly convert
-			// to int before passing to the %X formatter or else it throws.
-			sb.append(String.format("\\u%04X", (int)Character.highSurrogate(codePoint)));
-			sb.append(String.format("\\u%04X", (int)Character.lowSurrogate(codePoint)));
-		}
-		else {
-			sb.append(String.format("\\u%04X", codePoint));
-		}
+public class UnicodeEscapes {
+	public static String escapeCodePoint(int codePoint, TargetType targetType) {
+		StringBuilder result = new StringBuilder();
+		appendEscapedCodePoint(result, codePoint, targetType);
+		return result.toString();
 	}
 
-	static public void appendPythonStyleEscapedCodePoint(int codePoint, StringBuilder sb) {
-		if (Character.isSupplementaryCodePoint(codePoint)) {
-			sb.append(String.format("\\U%08X", codePoint));
-		}
-		else {
-			sb.append(String.format("\\u%04X", codePoint));
+	public static void appendEscapedCodePoint(StringBuilder sb, int codePoint, TargetType targetType) {
```

Comment:
This looks a lot more inconvenient but it appears that we need more flexibility; is that correct? In another words, it looks like you're saying that different targets need to generate different things. I wonder why that was working before.

Reply:
Different targets may have different string literal syntax (escaping), but some of them have the same syntax. It's ok that some branches are common for different targets.

Reply:
This method does not need fixing because it's used in [Target.appendUnicodeEscapedCodePoint](https://github.com/KvanTTT/antlr4/blob/tool-bugfixes/tool/src/org/antlr/v4/codegen/Target.java#L173) that can be easily overridden in unofficial targets if it's needed.

Reply:
We do not need this functionality now so you are introducing complexity and flexibility that we don't currently need. I really think there's a large principle here that you never change code unless you have to for fear of breaking something.  When you remove the enumerated type, this issue will go away I guess.

Reply:
> We do not need this functionality now so you are introducing complexity and flexibility that we don't currently need.

I just noted that such functionality had been implemented before my changes (other targets overrides `appendUnicodeEscapedCodePoint`) and my changes don't break this.